p2p/subscriber.go

```diff
@@ -100,23 +110,34 @@ func (s *Subscriber[H]) SetVerifier(val func(context.Context, H) error) error {
 			log.Errorw("invalid header",
 				"from", p.ShortString(),
 				"err", err)
+			s.metrics.observeReject(ctx)
 			return pubsub.ValidationReject
 		}
-		// keep the valid header in the msg so Subscriptions can access it without
-		// additional unmarhalling
-		msg.ValidatorData = hdr
 
 		var verErr *header.VerifyError
 		err = val(ctx, hdr)
 		switch {
-		case err == nil:
-			return pubsub.ValidationAccept
-		case errors.As(err, &verErr) && verErr.SoftFailure:
-			return pubsub.ValidationIgnore
 		default:
+			s.metrics.observeReject(ctx)
 			return pubsub.ValidationReject
+		case errors.As(err, &verErr) && verErr.SoftFailure:
+			s.metrics.observeIgnore(ctx)
+			return pubsub.ValidationIgnore
+		case err == nil:
+		}
+
+		now := time.Now()
+		if !lastAccept.IsZero() {
+			s.metrics.observeAccept(ctx, now.Sub(lastAccept), len(msg.Data))
```

Comment:
* This is feature request from Rollkit and they wanna measure propogation time for a size. I want this to see if we have any divergencies/anomalies in header sizes.
* I expect to have different size for every possible DAH, so not too much, but we need to see, because of Proto serialization that might not be constant size. I will test thing on Mocha before we decide

Reply:
If rollkit / we want to see different sizes attribute is not a way to go. It would cause cardinality issues and I don't think it would be possible to query properly from prometheus.  Alternative could be a histogram, where size is reported value. The trick would be to set up histogram buckets tho, which is tricky with current state of otel